src/compiler/program.ts

```diff
@@ -2702,6 +2707,34 @@ export function createProgram(rootNamesOrOptions: readonly string[] | CreateProg
         return hasEmitBlockingDiagnostics.has(toPath(emitFileName));
     }
 
+    function getCustomTransformers() {
+        if (!host.require) {
+            return emptyArray;
+        }

```

Comment:
maybe allow CompilerOptions to feed transformer factories directly, it will be better for API users.